### DISCUSSION TITLE
Access is denied. (Exception from HRESULT: 0x80070005 (E_ACCESSDENIED))

### DISCUSSION THREAD

**Discussion by TitaAE** [pnp/powershell#3894] [General]
$clientID = ""
$clientSc = ""
$TenantId = ""

#Parameters
$SiteURL = "https://testsite/"
$ListName = "ListName"
#Connect to SharePoint Online
Connect-PnPOnline -Url $SiteURL -ClientId $clientID -ClientSecret $clientSc

Get-PnPList | Select Title, ItemCount

Throws error in subject, but I have all permissions set, can someone point what is wrong?
![image](https://github.com/pnp/powershell/assets/161860529/531c7f92-b217-4d96-93e3-53a0d12637a2)

**Comment by AndersRask** [comment]
When you use secrets instead of certificates, it means that you are using legacy authentication ACS (Access Control Services). Legacy auth uses the XML based permission grants with SharePoint app-only service principals.
https://learn.microsoft.com/en-us/sharepoint/dev/solution-guidance/security-apponly-azureacs

If your permissions above are granted correctly, try adding a certificate to the app registration instead of a secret, and use the cert thumbnail or base64 encoded cert, and it should work.
More info
https://pnp.github.io/powershell/articles/authentication.html
